resolwe_bio/processes/variant_calling/gatk_variant_filtration.py

```diff
@@ -27,10 +37,173 @@ class GatkVariantFiltration(Process):
+        build = StringField(label="Build")
+
+    def run(self, inputs, outputs):
+        """Run analysis."""
+
+        TMPDIR = os.environ.get("TMPDIR")
+
+        filtered_variants = "filtered_variants.vcf.gz"
+        filtered_variants_index = filtered_variants + ".tbi"
+
+        # check the VCF file content
+        sample_count = return_sample_count(
+            vcf=inputs.vcf.output.vcf.path, error=self.error
+        )
+        if not sample_count > 1:
+            self.error(
+                f"The input VCF file should contain data for multiple samples. "
+                f"The input contains data for {sample_count} sample(s)."
+            )
+
+        if len(inputs.filter_expressions) != len(inputs.filter_name):
+            self.error(
+                "The number of filter expressions and filter names is not the same."
+            )
+
+        gc_threads = min(
+            self.requirements.resources.cores, inputs.advanced.java_gc_threads
+        )
+
+        args = [
+            "--java-options",
+            f"-XX:ParallelGCThreads={gc_threads} -Xmx{inputs.advanced.max_heap_size}g",
+            "-V",
+            inputs.vcf.output.vcf.path,
+            "-R",
+            inputs.ref_seq.output.fasta.path,
+            "-O",
+            filtered_variants,
+            "--window",
+            inputs.advanced.window,
+            "--cluster",
+            inputs.advanced.cluster,
+            "--tmp-dir",
+            TMPDIR,
+        ]
+
+        for name, exp in zip(inputs.filter_name, inputs.filter_expressions):
+            args.extend(["--filter-name", name, "--filter-expression", exp])
+
+        return_code, stdout, stderr = Cmd["gatk"]["VariantFiltration"][args] & TEE(
+            retcode=None
+        )
+        if return_code:
+            print(stdout, stderr)
+            self.error(
+                "GATK VariantFiltration failed. Check standard output for more "
+                "information."
+            )
+
+        outputs.vcf = filtered_variants
+        outputs.tbi = filtered_variants_index
+        outputs.species = inputs.vcf.output.species
+        outputs.build = inputs.vcf.output.build
+
+
+class GatkVariantFiltrationSingle(Process):
+    """Filter single-sample variant calls based on INFO and/or FORMAT annotations.
+
+    This tool is designed for hard-filtering variant calls based on certain criteria.
+    Records are hard-filtered by changing the value in the FILTER field to something
+    other than PASS. Passing variants are annotated as PASS and failing variants are
+    annotated with the name(s) of the filter(s) they failed. If you want to remove
+    failing variants, use GATK SelectVariants process.
+    """
+
+    slug = "gatk-variant-filtration-single"
+    process_type = "data:variants:vcf:variantfiltration:single"
+    name = "GATK VariantFiltration (single-sample)"
+    version = "1.0.0"
+    entity = {
+        "type": "sample",
+    }
```

Comment:
Data name for this process could also be something like `data_name = "Filtered variants ({{ vcf|sample_name|default("?") }})"`